client/src/editor.ts

```diff
@@ -0,0 +1,66 @@
+import { ExtensionContext, Range, SnippetString, TextDocument, commands, window } from "vscode";
+
+export function initialise(context: ExtensionContext) {
+	context.subscriptions.push(
+		commands.registerCommand("vscode-rpgle.surroundWithMonitor", () => surroundWithSnippet(lines => [
+			"",
+			"monitor;",
+			...lines,
+			"on-excp '$1';",
+			"on-error$2;",
+			"endmon;",
+			""
+		]))
+	);
+}
+
+function surroundWithSnippet(snippetProvider: (lines: string[]) => string[]) {
+	const editor = window.activeTextEditor;
+	if (editor) {
+		const document = editor.document;
+		let line = editor.selection.start.line;
+		if (isFreeRPGLE(document, line)) {
+			// First let's find out if this line starts elsewhere
+			while ((line - 1) >= 0 && !document.lineAt(line).text.trim().endsWith(`;`) && line-- > 0);
+			const startLine = line;
+
+			// Now get all the relevant lines
+			const lines: string[] = [];
+			let initialIdentation = document.lineAt(line).firstNonWhitespaceCharacterIndex;
+			while (line < document.lineCount) {
+				const currentLine = document.lineAt(line++);
+				const text = currentLine.text;
+				//Keep identation
+				const indent = currentLine.firstNonWhitespaceCharacterIndex - initialIdentation;
+				lines.push(`${"".padEnd(2 + (indent > 0 ? indent : 0))}${text.trim()}`);
+				const maybeComment = text.lastIndexOf("//");
+				if (line > editor.selection.end.line && (maybeComment > -1 ? text.substring(maybeComment + 1) : text).endsWith(';')) {
+					break;
+				}
+			}
+
+			editor.insertSnippet(new SnippetString(snippetProvider(lines).join("\n")),
+				new Range(startLine, document.lineAt(startLine).firstNonWhitespaceCharacterIndex, line - 1, 9999));
+		}
+		else {
+			window.showWarningMessage("Selection is not valid Free RPGLE code");
+		}
+	}
+}
+
+function isFreeRPGLE(document: TextDocument, currentLine: number) {
+	if (document.lineAt(0).text.substring(0, 6).toLowerCase() === `**free`) {
+		return true;
+	}
+	else {
+		for (let i = currentLine; i < document.lineCount; i++) {
```

Comment:
I am not sure about this check. In modern versions of RPGLE (7.3+), `/free` and `/end-free` are optional, so if they're not used, then it might always return `false` by the looks of it.